### PR TITLE
145 warning overwrite

### PR DESCRIPTION
**Pull request by maurolepore**

### CODE REVIEW

File: tests/testthat/test-match_name.R [hunk 1]
```diff
@@ -269,15 +269,27 @@ test_that("takes `p`", {
...
 
 test_that("takes `overwrite`", {
-  overwrite <- overwrite_demo
+  lbk <- slice(loanbook_demo, 4:25)
+
   expect_false(
     identical(
-      match_name(slice(loanbook_demo, 4:25), ald_demo, overwrite = NULL),
-      match_name(slice(loanbook_demo, 4:25), ald_demo, overwrite = overwrite)
+      match_name(lbk, ald_demo, overwrite = NULL),
```
Comment (maurolepore) [inline tests/testthat/test-match_name.R]: Ups, I should use `fake_*()` for speed.

File: tests/testthat/test-match_name.R [hunk 2]
```diff
@@ -269,15 +269,27 @@ test_that("takes `p`", {
...
+      suppressWarnings(match_name(lbk, ald_demo, overwrite = overwrite_demo))
```
Comment (maurolepore) [inline tests/testthat/test-match_name.R]: Use `fake_*()` for speed.

File: R/restructure_loanbook_for_matching.R [hunk 3]
```diff
@@ -94,6 +94,16 @@ may_overwrite_name_and_sector <- function(data, overwrite) {
     return(data)
   }
 
+  warn(
+    message = glue(
+      "You should only overwrite a sector at the level of the 'direct
+      loantaker' (DL). If you overwrite a sector at the level of the 'ultimate
+      parent' (UP) you consequently overwrite all children of that sector,
+      which most likely is a mistake."
```
Comment (maurolepore) [inline R/restructure_loanbook_for_matching.R]: @jdhoffa and @fpalandri, is this accurate and clear?

File: tests/testthat/test-match_name.R (hunk 1 above)
Comment (maurolepore) [reply tests/testthat/test-match_name.R]: Oh, I'm using a small slice anyway so speed isn't a problem.

File: tests/testthat/test-match_name.R (hunk 2 above)
Comment (maurolepore) [reply tests/testthat/test-match_name.R]: I'm using a small slice, so no speed problem.

File: R/restructure_loanbook_for_matching.R (hunk 3 above)
Comment (jdhoffa) [reply R/restructure_loanbook_for_matching.R]: I think at this stage this is clear! until we think of a way for this to not be an issue

**Pull request merged**